### PR TITLE
pythonPackages.dask-gateway: init at 0.8.0

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

Looking at creating dask-gateway nixos service for hpc computations. Need package to exist first.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 94693` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
<br>- python37Packages.dask-gateway
<br>- python37Packages.dask-gateway-server
<br>- python38Packages.dask-gateway
<br>- python38Packages.dask-gateway-server
</details>

**Comment by costrouc** [comment]
@GrahamcOfBorg build python3Packages.dask-gateway python3Packages.dask-gateway-server

File: pkgs/development/python-modules/dask-gateway/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, buildPythonPackage
+, fetchPypi
+, aiohttp
+, dask
+, distributed
+}:
+
+buildPythonPackage rec {
+  pname = "dask-gateway";
+  # update dask-gateway lock step with dask-gateway-server
```
Comment (jonringer) [inline pkgs/development/python-modules/dask-gateway/default.nix:11]: there should be a similar comment in the other package

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -2733,6 +2733,10 @@ in {
 
   dask = callPackage ../development/python-modules/dask { };
 
+  dask-gateway = callPackage ../development/python-modules/dask-gateway { };
+
+  dask-gateway-server = callPackage ../development/python-modules/dask-gateway-server { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:2738]: future proof from a python3Packages.go package being added
```suggestion
  dask-gateway-server = callPackage ../development/python-modules/dask-gateway-server {
    inherit (pkgs) go;
  };
```

**Review by jonringer — COMMENTED** [review]
if the server is just meant to be an application (E.g: `dask-server ...` ) I would prefer for it to be packaged as an application. Otherwise the use of go needs to be patched into the call sites if it's a package.

**Review by FRidh — COMMENTED** [review]
In case they're needed to be updated in lockstep, consider adding the other package to `passthru.tests`. At least ofborg will test that then.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (costrouc) [reply pkgs/top-level/python-packages.nix]: done

File: pkgs/development/python-modules/dask-gateway/default.nix (hunk 1 above)
Comment (costrouc) [reply pkgs/development/python-modules/dask-gateway/default.nix:11]: done

**Comment by costrouc** [comment]
@jonringer needs to be a python package due to how it is used. Often times the `dask-gateway-server` is subclassed to add additional features to the server. I wish that they hadn't decided to build a go proxy server that is self contained in the package (the proxy server will not be extended). Especially since there are existing tools that fit their use case e.g. traefik. 

@FRidh both of these packages come from the same git repo https://github.com/dask/dask-gateway. So me saying that they need to be updated together is because they are released together. Should I switch to using the git repo and build from that? Should I set `dask-gateway.src = dask-gateway-server.src; dask-gateway.version = dask-gateway-server.version;` for example. I'm thinking this would be more robust.

**Comment by costrouc** [comment]
@GrahamcOfBorg build python3Packages.dask-gateway python3Packages.dask-gateway-server

File: pkgs/development/python-modules/dask-gateway/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  propagatedBuildInputs = [
+    aiohttp
+    dask
+    distributed
+  ];
+
+  # tests not included with sdist and requires cluster for testing
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/dask-gateway/default.nix:26]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

File: pkgs/development/python-modules/dask-gateway/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with lib; {
+    description = "A client library for interacting with a dask-gateway server";
+    homepage = https://gateway.dask.org/;
```
Comment (jonringer) [inline pkgs/development/python-modules/dask-gateway/default.nix:30]: rfc 45
```suggestion
    homepage = "https://gateway.dask.org/";
```

File: pkgs/development/python-modules/dask-gateway-server/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  preBuild = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  # tests not included with sdist and requires cluster for testing
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A multi-tenant server for securely deploying and managing multiple Dask clusters";
+    homepage = https://gateway.dask.org/;
```
Comment (jonringer) [inline pkgs/development/python-modules/dask-gateway-server/default.nix:43]:
```suggestion
    homepage = "https://gateway.dask.org/";
```

File: pkgs/development/python-modules/dask-gateway/default.nix (hunk 4 above)
Comment (costrouc) [reply pkgs/development/python-modules/dask-gateway/default.nix]: done.

File: pkgs/development/python-modules/dask-gateway-server/default.nix (hunk 5 above)
Comment (costrouc) [reply pkgs/development/python-modules/dask-gateway-server/default.nix]: done.

File: pkgs/development/python-modules/dask-gateway/default.nix (hunk 3 above)
Comment (costrouc) [reply pkgs/development/python-modules/dask-gateway/default.nix:26]: added pythonInputsCheck to packages

Comment (costrouc) [reply pkgs/development/python-modules/dask-gateway/default.nix:26]: since tests required a running slurm cluster/yarn/etc.

**Comment by costrouc** [comment]
Result of `nixpkgs-review pr 94693` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
<br>- python37Packages.dask-gateway
<br>- python37Packages.dask-gateway-server
<br>- python38Packages.dask-gateway
<br>- python38Packages.dask-gateway-server
</details>

**Comment by costrouc** [comment]
@GrahamcOfBorg build python3Packages.dask-gateway python3Packages.dask-gateway-server

**Comment by jonringer** [comment]
do you mind reviewing #94894

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 94693 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  - python37Packages.dask-gateway  - python37Packages.dask-gateway-server  - python38Packages.dask-gateway  - python38Packages.dask-gateway-server</details>

**Pull request merged**